scripts/debug_tools/reduce_repro_files.py

```diff
@@ -0,0 +1,107 @@
+import argparse
+import re
+import os
+import shutil
+import stat
+import subprocess
+import multiprocessing
+
+
+def get_file_path(analyzer_command_file):
+    with open(analyzer_command_file, 'r') as f:
+        return f.read().split(" ")[-1]
+
+
+def get_preprocessed_repro_file(abs_file_path, analyzer_command_file):
+    with open(analyzer_command_file, 'r') as f:
+        cmd = f.read().split(" ")
+        param_pattern = re.compile("-I|-D")
+        prepoc_params = [x for x in cmd if param_pattern.match(x)]
+        preproc = \
+            subprocess.Popen(["gcc", "-E"] + prepoc_params + [abs_file_path],
+                             stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        out, _ = preproc.communicate()
+        filename = abs_file_path.split('/')[-1]
+        # assuming exactly one dot (.) in the file name
+        prepoc_name = filename.split('.')[0] + "_preproc." + \
+            filename.split('.')[1]
+    with open(prepoc_name, 'w') as preproc_file:
+        preproc_file.write(out)
+    return prepoc_name
+
+
+def get_assertion_string(analyzer_command_file):
+    error = subprocess.Popen(["bash", analyzer_command_file],
+                             stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    _, error_string = error.communicate()
+    assert_pattern = re.compile('Assertion.+failed\.')
+    assert_match = assert_pattern.search(error_string)
+    if not assert_match:
+        return ""
+    return assert_match.group(0)
+
+
+def reduce(prepoc_name, assert_string, analyzer_command_file, num_threads):
+    reduce_file_name = prepoc_name.replace("preproc", "reduce")
+    if not os.path.exists(reduce_file_name):
+        shutil.copy2(prepoc_name, reduce_file_name)
+    conditions = []
+    compilable_cond = ['gcc', '-c', '-Werror', reduce_file_name]
```

Comment:
Why do we use gcc here? It does not compile the same set of inputs as clang. Does it make sense to use `-Werror`?